src/core/core.c

```diff
@@ -189,26 +202,44 @@ void core_register_options(void)
 
 void core_preinit(const char *path)
 {
+	struct stat statbuf;
 	const char *home;
 	char *str;
 	int len;
+	int use_xdg = 0;
 
 	if (irssi_dir == NULL) {
-		home = g_get_home_dir();
-		if (home == NULL)
-			home = ".";
-
-		irssi_dir = g_strdup_printf(IRSSI_DIR_FULL, home);
+		/* check if %XDG_CONFIG_HOME/irssi exists and use to it */
+		char *dirp = g_build_filename(g_get_user_config_dir(), "irssi", NULL);
+		if (stat(dirp, &statbuf) == 0) {
```

Comment:
```suggestion
		if (!g_file_test(dirp, G_FILE_TEST_EXISTS)) {
```

And `dirp` is leaked if the`if` body is executed.